### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by pedro-lima-jalafund**
[user story #29](https://dev.azure.com/funjaladev/DevLevel3/_boards/board/t/DevLevel3%20Team/Stories/?workitem=29)

### CODE REVIEW

**Comment by fernando-maceda-jalafund** [comment]
I think you already have a base project, so you may want to include linting and building process on your pipeline config file

**Comment by martin-callejas-jalafund** [comment]
> I think you already have a base project, so you may want to include linting and building process on your pipeline config file

AC got trimmed for this sprint, so only build and drop is required atm

**Comment by martin-callejas-jalafund** [comment]
please add the link to the user story in the description

**Comment by martin-callejas-jalafund** [comment]
rename the branch to follow conventions
you may need to open a new PR

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
+# Add steps that build, run tests, deploy, and more:
+# https://aka.ms/yaml
+
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:2]: let's delete all unnecessary comments

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+trigger:
+  # batch runs this pipeline one after the other (useful when there are many contributors)
+  batch: true
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:5]: I'm not sure about the batch functionality nor used it before. If this some kind of extra-mile please let's push it to our tech debt backlog and discuss if it's needed

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  branches:
+    include:
+      - master
+      - develop
+
+pr:
+  autoCancel: true
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:12]: default is true so line 15 is not needed https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#multiple-pr-updates

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+trigger:
+  # batch runs this pipeline one after the other (useful when there are many contributors)
+  batch: true
+  branches:
+    include:
+      - master
+      - develop
+
+pr:
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:11]: let's set drafts=false so we won't get PR builds for drafts PRs https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#draft-pr-validation

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    # installing dependencies (react, axios, etc)
+    - script: npm install
+
+    # building the webApp
+    - script: npm run build
+    #for conditional run of a step or job or stage
+
+    
+    - task: PublishBuildArtifacts@1
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:41]: indentation is important for markdown readers so let's keep a 2 spaces indentation

File: azure-pipelines.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  branches:
+    include:
+      - master
+      - develop
+
+pool:
+  vmImage: ubuntu-latest
+
+stages:
+- stage: Initial
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:22]: let's keep it simple, no stages atm: just a single step and the tasks

File: azure-pipelines.yml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  jobs:
+  - job: Initializing
+    steps:
+    - task: NodeTool@0
```
Comment (martin-callejas-jalafund) [inline azure-pipelines.yml:26]: please set a meaningful displayName to this task

File: test.js [hunk 8]
```diff
@@ -0,0 +1 @@
+console.log("testing the web app pipeline")
```
Comment (martin-callejas-jalafund) [inline test.js:1]: please delete this file

**Review by martin-callejas-jalafund — CHANGES_REQUESTED** [review]
awesome starter work with extra-miles! let's refactor to have it simpler and shorter

File: azure-pipelines.yml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      inputs:
+        versionSpec: '14.17.x' 
+      #force32bit: false # Optional
+      #checkLatest: false # Optional #version LTS o latest de node
+
+    # installing dependencies (react, axios, etc)
+    - script: npm install
```
Comment (DaleneyRuben) [inline azure-pipelines.yml:32]: It'd be better to run [npm ci](https://docs.npmjs.com/cli/v7/commands/npm-ci)

File: azure-pipelines.yml (hunk 1 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml:2]: done

File: azure-pipelines.yml (hunk 2 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml:5]: can you please point me to the right place where "tech debt backlog" is?

File: azure-pipelines.yml (hunk 3 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml:12]: Yes, it was only for studies purposes, but I will remove it

File: azure-pipelines.yml (hunk 4 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml:11]: done

File: azure-pipelines.yml (hunk 6 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml:22]: ok

File: azure-pipelines.yml (hunk 7 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml:26]: done

File: test.js (hunk 8 above)
Comment (pedro-lima-jalafund) [reply test.js:1]: done

File: azure-pipelines.yml (hunk 9 above)
Comment (pedro-lima-jalafund) [reply azure-pipelines.yml:32]: Documentation readed, Done

**Pull request closed without merging**